python/lbann/launcher/batch_script.py

```diff
@@ -0,0 +1,168 @@
+
+    """
+
+    def __init__(self,
+                 script_file=None,
+                 work_dir=os.getcwd(),
+                 interpreter='/bin/bash'):
+        """Construct batch script manager.
+
+        Args:
+            script_file (str): Script file.
+            work_dir (str, optional): Working directory
+                (default: current working directory).
+            interpreter (str, optional): Script interpreter
+                (default: /bin/bash).
+
+        """
+
+        # Lines in script are stored as lists of strings
+        self.header = []
+        self.body = []
+
+        # Construct file paths
+        self.work_dir = os.path.realpath(work_dir)
+        self.script_file = script_file
+        if not self.script_file:
+            self.script_file = os.path.join(self.work_dir, 'batch.sh')
+        self.script_file = os.path.realpath(self.script_file)
+        self.out_log_file = os.path.join(self.work_dir, 'out.log')
+        self.err_log_file = os.path.join(self.work_dir, 'err.log')
+
+        # Shebang line
+        if interpreter:
+            self.header.append('#!{}'.format(interpreter))
+
+    def add_header_line(self, line):
+        """Add line to script header.
+
+        The header should specify configuration options for the job
+        scheduler, without containing executable commands.
+
+        """
+        self.header.append(line)
+
+    def add_body_line(self, line):
+        """Add line to script body.
+
+        The body should contain the script's executable commands.
+
+        """
+        self.body.append(line)
+
+    def add_command(self, command):
+        """Add executable command to script."""
+        self.add_body_line(command)
+
+    def add_parallel_command(self,
+                             command,
+                             launcher=None,
+                             launcher_args=None,
+                             nodes=None,
+                             procs_per_node=None):
+        """Add command to be executed in parallel.
+
+        The command is executed via a launcher, e.g. `mpirun`.
+        Parallel processes are distributed evenly amongst the compute
+        nodes.
+
+        Args:
+            command (str): Command to be executed in parallel.
+            launcher (str, optional): Parallel command launcher,
+               `mpirun`.
+            launcher_args (`Iterable` of `str`s, optional):
+                Command-line arguments to parallel command launcher.
+            nodes (int, optional): Number of compute nodes.
+            procs_per_node (int, optional): Number of parallel
+                processes per compute node.
+
+        """
+        raise NotImplementedError(
+            'classes that inherit from `BatchScript` should implement '
+            '`add_parallel_command` to use a specific job scheduler'
+        )
+
+    def write(self):
+        """Write script to file.
+
+        Overwrites script file if it already exists. The working
```

Comment:
It might make things slightly more flexible / avoid unfortunate failures if overwriting was optional (maybe default to not?) and controlled via a kwarg.

Reply:
Done.